cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/ClusterBootstrap.scala

```diff
@@ -33,13 +37,28 @@ final class ClusterBootstrap(implicit system: ExtendedActorSystem) extends Exten
   val settings = ClusterBootstrapSettings(system.settings.config)
 
   // used for initial discovery of contact points
-  val discovery: ServiceDiscovery = {
-    val clazz = settings.contactPointDiscovery.discoveryClass
-    system.dynamicAccess.createInstanceFor[ServiceDiscovery](clazz, List(classOf[ActorSystem] → system)).get
-  }
+  val discovery: SimpleServiceDiscovery =
+    settings.contactPointDiscovery.discoveryMethod match {
+      case "akka.discovery" ⇒
+        val discovery = ServiceDiscovery(system).discovery
+        log.info("Bootstrap using default `akka.discovery` mechanism: {}", Logging.simpleName(discovery))
+        discovery
+
+      case otherDiscoveryMechanism ⇒
+        val implClazz = system.settings.config.getString(otherDiscoveryMechanism + ".impl")
+        log.info("Bootstrap using [{}] discovery mechanism, instantiating [{}]", otherDiscoveryMechanism, implClazz)
+        system.dynamicAccess
+          .createInstanceFor[SimpleServiceDiscovery](implClazz, List(classOf[ActorSystem] → system))
+          .get
+    }
 
   private[this] val _selfContactPointUri: Promise[Uri] = Promise()
 
+  override def routes(routeProviderSettings: ManagementRouteProviderSettings): Route = {
+    this.setSelfContactPoint(routeProviderSettings.selfBaseUri)
```

Comment:
This is the reason we do the `ManagementRouteProviderSettings` dance.
We need this in order to know if "we are the lowest address" or not, since we do this decision on the management addresses which are from discovery